### PR TITLE
Update NEML to newest version

### PR DESCRIPTION
**Pull request by reverendbedford**
I finally eliminated error codes in NEML, so the interface here needs to change slightly.  This PR:

- Updates NEML to the current dev
- Uses the new NEMLError system to provide a more informative mooseError in the event of a failed stress update
- Updates the LANLTi regression test -- we keep messing with this model, sorry
- Updates the script that gathers the tests to avoid pulling in the `__pycache__` directories.  The NEML tests are now a proper python module.

### CODE REVIEW

File: src/materials/NEMLStressBase.C
Comment (dschwen) [inline src/materials/NEMLStressBase.C:155]: Updated?

Comment (reverendbedford) [reply src/materials/NEMLStressBase.C:155]: Ugh.

**Comment by dschwen** [comment]
This is nice. I'll approve this (if you want to fix the typo) whenever you're ready.

**Comment by reverendbedford** [comment]
Thanks, yes I fixed the typo just now.

**Review by dschwen — APPROVED** [review]
:+1:

**Comment by bwspenc** [comment]
Any reason we shouldn't merge this? I see that the 'updated' typo is still there in this version, but other than that, it looks fine.

**Comment by dschwen** [comment]
Looks like @reverendbedford just forgot to push. I'll merge now. This can be fixed later.

**Pull request merged**
